refactor!: remove 'remap' and 'terse' options

dundargoc opened the pull request:
remap:
An option that should always be kept on except for when working with old
Vi scripts is of little use to us.

terse:
Having two methods of setting the same option is redundant and can be
potentially confusing. Recommend adding 's' to 'shortmess' instead.

dundargoc committed:
refactor!: remove 'remap' option

An option that should always be kept on except for when working with old
Vi scripts is of little use to us.

gpanders commented on src/nvim/options.lua:
```diff
@@ -1909,13 +1909,6 @@ return {
-      type='bool', scope={'global'},
-      varname='p_remap',
-      defaults={if_true=true}
-    },
```
Follow the pattern used for e.g. `'compatible'`: change `varname` to `'p_force_off'` and the descriptino to `"No description"`.

gpanders commented on src/nvim/options.lua:
```diff
@@ -1911,9 +1911,9 @@ return {
+      short_desc=N_("No description"),
       type='bool', scope={'global'},
-      varname='p_remap',
+      varname='p_remap_off',
```
```suggestion
      varname='p_force_on',
```

dundargoc replied:
?

dundargoc replied:
Never mind, I cannot read. Thanks.

dundargoc committed:
refactor!: remove 'terse' option

Having two methods of setting the same option is redundant and can be
potentially confusing. Recommend adding 's' to 'shortmess' instead.

gpanders commented on runtime/doc/vim_diff.txt:
```diff
@@ -554,6 +554,7 @@ Options:
   *'shortname'* *'sn'* *'noshortname'* *'nosn'*
   *'swapsync'* *'sws'*
   *'termencoding'* *'tenc'* (Vim 7.4.852 also removed this for Windows)
+  *'terse'* *'noterse'* (Add 's' to *shortmess* instead)
```
Single quotes is used for options.

```suggestion
  *'terse'* *'noterse'* (Add "s" to 'shortmess' instead)
```

gpanders commented on runtime/doc/map.txt:
```diff
@@ -796,9 +796,8 @@ command.  For example: >
-With the normal :map command, when the 'remap' option is on, mapping takes
-place until the text is found not to be a part of a {lhs}.  For example, if
-you use: >
+With the normal :map command a mapping takes place until the text is found not
```
Inserted an extra "a"

```suggestion
With the normal :map command mapping takes place until the text is found not
```

zeertzjq reviewed (approved):
The `'terse'` option isn't even synchronized with `s` flag in `'shortmess'`, so `&terse` cannot even be used to check the current behavior.
